llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -543,6 +543,8 @@ static Instruction *foldCttzCtlz(IntrinsicInst &II, InstCombinerImpl &IC) {
       auto *Cttz = IC.Builder.CreateBinaryIntrinsic(Intrinsic::cttz, X,
                                                     IC.Builder.getTrue());
       auto *ZextCttz = IC.Builder.CreateZExt(Cttz, II.getType());
+      if (auto *ZextInst = dyn_cast<PossiblyNonNegInst>(ZextCttz))
+        ZextInst->setNonNeg();
```

Comment:
Is this correct for i1?

Reply:
It is correct for i1.
Alive2: https://alive2.llvm.org/ce/z/CvSF2z